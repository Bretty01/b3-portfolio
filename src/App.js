import logo from './logo.svg';
import './App.scss';
import {useRef} from "react"
import SkillStamp from "./components/SkillStamp/SkillStamp"
import Navigation from "./components/Navigation/Navigation"
function App() {

    const handleScroll = event => {
        console.log(event)
    }

  const navRefs = [
      useRef(null),
      useRef(null),
      useRef(null)
  ]

  return (
    <div className="main">
        <Navigation navrefs={navRefs}/>
        <div className="landing">
            <div>
                <h1>Brett Orban<strong>.</strong></h1>
                <h2>Success Through Hard Work and Dedication!</h2>
            </div>
            <h3>Full Stack Developer <strong>|</strong> Software Developer <strong>|</strong> Fun Guy</h3>
        </div>
        <div className="main-content">
            <div id="about" ref={navRefs[0]}>
                <h2>The name is <strong>Brett</strong>! Welcome aboard!</h2>
                <span>I'm a <strong>Full Stack Developer</strong> born and raised in Regina, Saskatchewan. While I
                have grown up with a love for technology, I have further appreciated the art of web and software
                through both the creativity and problem solving that programming offers.<br /><br />

                I am always excited to learn new skills and refine my existing skills to further improve myself and
                the people around me. Take a look around and see what I have to offer. Maybe you will get a good
                impression and be inspired!
                </span>
            </div>
            <div id="skills" ref={navRefs[1]}  onScroll={handleScroll}>
                <div>
                    <div className="tab-head">
                        <button>Frontend</button>
                        <button>Backend</button>
                        <button>Languages</button>
                        <button>Tools</button>
                    </div>
                    <div className="tab-body">
                        <SkillStamp title="React" image="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg" />
                        <SkillStamp title="React" image="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg" />
                        <SkillStamp title="React" image="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg" />
                        <SkillStamp title="React" image="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg" />
                        <SkillStamp title="React" image="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg" />
                        <SkillStamp title="React" image="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg" />
                        <SkillStamp title="React" image="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg" />
                    </div>
                </div>
            </div>
            <div id="projects" ref={navRefs[2]}>
                <h1>Projects</h1>
                <div>

                </div>
            </div>
        </div>
    </div>
  );
}

export default App;
