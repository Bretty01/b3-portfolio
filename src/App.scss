.main {
  background-color: #000000;
  min-height: 100vh;
  color: white;
  overflow-x: hidden;
}
.landing {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  > div {
    display: flex;
    flex-direction: row;

    h2 {
      max-width: 200px;
    }
  }

  .social-icons {
    position: absolute;
    top: 10px;
    right: 10px;
    a {
      color: white;
      &:visited{
        color: white;
      }
    }
    svg {
      height: 36px;
      width: 36px;
      padding: 0 5px;
      &:hover {
        fill: var(--accent-secondary);
      }
      &:hover + div{
        opacity: 1;
      }
    }
    > a{
      display: flex;
      flex-direction: column;
      align-items: flex-end;
      width: 46px;
      text-decoration: none;
    }
  }
}

#about {
  min-height: 100vh;
  > div {
    padding-top: 10vh;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    > * {
      width: 45%
    }
  }
}

.landing {
  padding: 0 10%;
}

#main-content {
  width: 80%;
  margin-left: auto;

  > div {
    min-height: 100vh;
    padding: 0 5%;
  }

  h1 {
    text-align: center;
  }
}

#skills {
    > div {
      margin: 0 10%;
      min-height: 300px;
      border: 1px solid white;
    }
}

.tab-head {
  button {
    background-color: black;
    color: white;
    width: 25%;
    height: 80px;
    border: 1px solid white;
  }
}
.tab-body{
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
}

#projects {
  display: flex;
  flex-direction: column;
  .projects-link {
    font-size: 1.8em;
    text-align: center;
  }
}

form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;

  > * {
    background-color: black;
    color: lightgrey;
    border-color: lightgrey;
  }

  input, textarea {
    padding: 1%;
    width: 80%;
    margin: 5px auto 5px auto;
  }
  input[type="submit"] {
    width: 30%;
  }
  textarea {
    height: 300px;
  }
}
.speech-bubble {
  position: relative;
  background: var(--accent-secondary);
  border-radius: .4em;
  top: 10px;
  right: 8px;
  width: max-content;
  padding: 0.25em 0.5em;
  opacity: 0;
  transition: opacity 500ms;
}

.speech-bubble:after {
  content: '';
  position: absolute;
  top: 0;
  right: 15px;
  width: 0;
  height: 0;
  border: 7px solid transparent;
  border-bottom-color: var(--accent-secondary);
  border-top: 0;
  border-right: 0;
  margin-left: -3.5px;
  margin-top: -7px;
}
