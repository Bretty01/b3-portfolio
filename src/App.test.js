import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import App from './App';

test('App snapshots', () => {
  const tree = render(<App />)
  expect(tree).toMatchSnapshot()
  const arrowIcon = screen.getByText(/Click here to continue!/i)
  userEvent.click(arrowIcon)
  expect(tree).toMatchSnapshot()
})

test('App shows landing section', () => {
  render(<App />);
  const linkElement = screen.getByText(/Brett Orban/i);
  expect(linkElement).toBeInTheDocument();
});

test('App does not show main section', () => {
  render(<App />)
  const unshownElement = screen.queryByText(/Projects/i)
  expect(unshownElement).toBeNull()
})

test('click ArrowDownwardIcon to show main section', () => {
  render(<App />)
  const arrowIcon = screen.getByText(/Click here to continue!/i)
  userEvent.click(arrowIcon)
  const unshownElement = screen.queryAllByText(/Projects/i)
  expect(unshownElement).not.toBeNull()
})


