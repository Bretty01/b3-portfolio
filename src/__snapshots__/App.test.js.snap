// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="scrollhost-container"
      >
        <div
          class="scrollhost undefined"
        >
          <div
            class="main"
          >
            <div
              class="landing"
              style="margin-bottom: 0px;"
            >
              <div
                class="landing-title"
              >
                <h2>
                  <strong>
                    Full Stack Developer
                  </strong>
                </h2>
                <h1>
                  Brett Orban
                </h1>
                <span>
                  <i>
                    <u>
                      It ain't a problem without a solution!
                    </u>
                  </i>
                </span>
              </div>
              <div
                id="display-main-content"
              >
                <span>
                  Click here to continue!
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDownwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </div>
              <div
                class="social-icons"
              >
                <a
                  href="https://github.com/Bretty01"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="GitHubIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                    />
                  </svg>
                  <div
                    class="speech-bubble"
                  >
                    Github
                  </div>
                </a>
                <a
                  href="https://www.linkedin.com/in/brett-orban-93994324a/"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="LinkedInIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                    />
                  </svg>
                  <div
                    class="speech-bubble"
                  >
                    LinkedIn
                  </div>
                </a>
                <a
                  href="/Brett Orban Resume.pdf"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArticleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-5 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
                    />
                  </svg>
                  <div
                    class="speech-bubble"
                  >
                    Resume
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="scroll-bar"
          style="opacity: 0;"
        >
          <div
            class="scroll-thumb"
            style="height: 20px; top: 0px;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="scrollhost-container"
    >
      <div
        class="scrollhost undefined"
      >
        <div
          class="main"
        >
          <div
            class="landing"
            style="margin-bottom: 0px;"
          >
            <div
              class="landing-title"
            >
              <h2>
                <strong>
                  Full Stack Developer
                </strong>
              </h2>
              <h1>
                Brett Orban
              </h1>
              <span>
                <i>
                  <u>
                    It ain't a problem without a solution!
                  </u>
                </i>
              </span>
            </div>
            <div
              id="display-main-content"
            >
              <span>
                Click here to continue!
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowDownwardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </div>
            <div
              class="social-icons"
            >
              <a
                href="https://github.com/Bretty01"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="GitHubIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                  />
                </svg>
                <div
                  class="speech-bubble"
                >
                  Github
                </div>
              </a>
              <a
                href="https://www.linkedin.com/in/brett-orban-93994324a/"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LinkedInIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                  />
                </svg>
                <div
                  class="speech-bubble"
                >
                  LinkedIn
                </div>
              </a>
              <a
                href="/Brett Orban Resume.pdf"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArticleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-5 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
                  />
                </svg>
                <div
                  class="speech-bubble"
                >
                  Resume
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="scroll-bar"
        style="opacity: 0;"
      >
        <div
          class="scroll-thumb"
          style="height: 20px; top: 0px;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
